app/uk/gov/hmrc/emcstfe/controllers/SubmitReportOfReceiptController.scala

```diff
@@ -30,15 +33,22 @@ import scala.concurrent.ExecutionContext
 @Singleton()
 class SubmitReportOfReceiptController @Inject()(cc: ControllerComponents,
                                                 service: SubmitReportOfReceiptService,
+                                                val config: AppConfig,
                                                 override val auth: AuthAction
-                                               )(implicit ec: ExecutionContext) extends BackendController(cc) with AuthActionHelper with Logging {
+                                               )(implicit ec: ExecutionContext) extends BackendController(cc) with AuthActionHelper with Logging with FeatureSwitching {
 
   def submit(ern: String, arc: String): Action[JsValue] = authorisedUserSubmissionRequest(ern) { implicit request =>
     withJsonBody[SubmitReportOfReceiptModel] { submission =>
-      service.submit(submission).map {
-        case Left(value) => InternalServerError(Json.toJson(value))
-        case Right(value) => Ok(Json.toJson(value))
+      if(isEnabled(SendToEIS)) {
+        service.submitViaEIS(submission).map(handleResponse(_))
+      } else {
+        service.submit(submission).map(handleResponse(_))
```

Comment:
Might be better to FS in the Service layer. As it's that which decides which connector to then use. Controller doesn't really need to care as method signature and response is the same.

Reply:
Response wasn't the same :( my mistake. Kept as is